docs/explanation/upstream-and-downstream.rst

```diff
@@ -1,9 +1,139 @@
+    directly or indirectly.
+
+    *Examples*
+        - :term:`Debian` is the upstream of :term:`Ubuntu`.
+        - *"Upstream is not interested in the patch."*
+
+    *Usage note*
+        - There can be many layers. For example, *Kubuntu* is a :term:`flavour <Ubuntu flavours>`
+          of :term:`Ubuntu`, therefore :term:`Ubuntu` and :term:`Debian` are upstreams of
+          *Kubuntu*.
+        - The adjective/adverb form is much more commonly used.
+
+.. _upstream_adjective_adverb:
+
+*upstream (adjective, adverb)*
+    Something (usually a code modification like a :term:`Patch`) that flows in the direction or is
+    relative to a software project closer to the original software project.
+
+    *Examples*
+        - :term:`Debian` is the upstream project of :term:`Ubuntu`.
+        - *"There is a new upstream release."*
+        - *"A pull request was created upstream."*
+        - *"A bug was patched upstream."*
+
+.. _upstream_verb:
+
+*upstream (verb)*
+    Sending something (usually a :term:`Patch`) upstream that originated from a :term:`Fork` or
+    project that depended on the upstream project.
+    
+    *Examples*
+        - *"We upstreamed the patch."*
+        - *"Can you upstream the bugfix?"*
+
+.. _Downstream:
+
+Downstream
+~~~~~~~~~~
+
+*downstream (noun)*
+    Similar to :ref:`upstream_noun`: A software project(s) (and associated entities) that depend on
+    another software project directly or indirectly.
+
+    *Example*
+        - :term:`Ubuntu` is a downstream of :term:`Debian` and there are many downstreams
+          of :term:`Ubuntu`.
+
+    *Usage note*
+        - The :ref:`adjective/adverb form <downstream_adjective_adverb>` is much more commonly used.
+        - There can be many layers. For example, *Kubuntu* is a :term:`flavour <Ubuntu flavours>`
+          of :term:`Ubuntu`, therefore *Kubuntu* and :term:`Ubuntu` are downstreams of
+          :term:`Debian`.
+
+.. _downstream_adjective_adverb:
+
+*downstream (adjective, adverb)*
+    Similar to :ref:`upstream_adjective_adverb`: Something (usually a code modification like 
+    a :term:`Patch`) that flows in the direction or is relative to a software project farther
+    away from the original software project.
+
+    *Examples*
+        - :term:`Ubuntu` is a downstream project of :term:`Debian`.
+        - *"The bug is already patched downstream."*
+        - *"The bug was reported by a downstream user."*
+        - *"Downstream maintainers have submitted a bugfix."*
+        - *"The change may affect downstream users."*
+
+*downstream (verb)*
+    Similar to :ref:`upstream_verb`: Sending something (usually a :term:`Patch`) downstream
+    that originated from an upstream project.
+
+    *Example*
+        - "We downstreamed the :term:`Patch`."
+
+Why upstream changes?
+---------------------
+
+.. note::
+    The following list does not aim for completeness. There are certainly more good arguments 
```

Comment:
```suggestion
    The following list does not aim for completeness. There are plenty more good arguments 
```

Reply:
("certainly more good" could be read like there are better arguments that aren't listed)